juju/juju
# feat(macaroons): Implement bakery config service

Implement bakery config service, backed by DQLite.

Create a SQLite bakced service to take the responsibility of:
https://github.com/juju/juju/blob/034ed117eb114090027104e448aeb9453b39d20d/state/bakerystorage/config.go

Expose the service to the service factory + initialise in bootstrap worker

## Checklist


## QA steps

```
go test github.com/juju/juju/domain/bakerystorage/...
```

Verify table initialised (following guide to open a DQLite repl from https://github.com/juju/juju/pull/16519)

```
$ juju bootstrap lxd lxd
$ make repl-install
$ juju ssh -m controller 0
$ sudo cat /var/lib/juju/agents/machine-0/agent.conf | yq '.controllercert' | xargs -I% echo % > dqlite.cert
$ sudo cat /var/lib/juju/agents/machine-0/agent.conf | yq '.controllerkey' | xargs -I% echo % > dqlite.key
$ sudo dqlite -s file:///var/lib/juju/dqlite/cluster.yaml -c ./dqlite.cert -k ./dqlite.key controller
dqlite> SELECT * FROM bakery_config
[204 210 ... 230 125]|[200 163 ... 69 49]|[52 64 ... 130 70]|[133 200 ... 212 126]|[13 170 ... 46 69]|[171 200 ... 192 17]|[134 136 ... 64 8]|[195 142 ... 133 16]
```

## Links

https://warthogs.atlassian.net/browse/JUJU-5856

[JUJU-5856]: https://warthogs.atlassian.net/browse/JUJU-5856?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
